.github/workflows/main.yml

```diff
@@ -4,14 +4,14 @@ jobs:
   job:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
       - uses: trilom/file-changes-action@v1.2.4
         id: file_changes
         with:
           output: ''
       - uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 2.4.1
+          ruby-version: 2.6
```

Comment:
In the official repository the [supported version of Ruby](https://github.com/ruby/setup-ruby/blob/51c0d5922252eab2cb7349f2142160df023fb656/README.md?plain=1#L17) are until `3.1.0` but in the [usage example](https://github.com/ruby/setup-ruby/blob/51c0d5922252eab2cb7349f2142160df023fb656/README.md?plain=1#L70) there is still `2.6`
Is it better to update the version to ` 3.1.0`?

Reply:
Ruby is used to check urls. Need check compatibility with that program

Reply:
In the [awesome_bot example](https://github.com/dkhamsing/open-source-ios-apps/blob/master/.github/workflows/ruby.yml) it uses `ruby-version: 2.6.x`, I don't know if newest version are compatible

Reply:
> In the [awesome_bot example](https://github.com/dkhamsing/open-source-ios-apps/blob/master/.github/workflows/ruby.yml?rgh-link-date=2022-01-26T13%3A24%3A32Z) it uses `ruby-version: 2.6.x`, I don't know if newest version are compatible

Reviewing a bit more in documentation (https://github.com/dkhamsing/awesome_bot/blob/4989c0f248be71089ad300d93dd0b2abb930337e/.github/workflows/ci.yml#L15) test matrix is done over Ruby 2.6, 2.7 and 3.0

I completelly unknown Ruby but I think it's safe upgrade its version.

Reply:
Maybe @dkhamsing can help us, what versions of ruby have compatibility with awesome_bot?

Reply:
awesome_bot ruby requirement is version 2, sorry if it's not better documented

Reply:
Thanks for the clarification

Reply:
@LuigiImVector then alter some handled `.md` file to see how CI works and check after the differences

Reply:
Seems that everything works correctly: [#2](https://github.com/LuigiImVector/free-programming-books/pull/2) (sorry for the confusing PR).

Reply:
Yep. It works. I imagine the amount of time elapsed to process all files :laughing: 

Workflow log on that PR: [logs_33.zip](https://github.com/EbookFoundation/free-programming-books/files/8063183/logs_33.zip)